File: android/src/main/java/com/fanhero/fanhero_native_utils/FanheroNativeUtilsPlugin.java
package com.fanhero.fanhero_native_utils;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/** FanheroNativeUtilsPlugin */
public class FanheroNativeUtilsPlugin implements MethodCallHandler {
  /** Plugin registration. */
  public static void registerWith(Registrar registrar) {
    final MethodChannel channel = new MethodChannel(registrar.messenger(), "fanhero_native_utils");
    channel.setMethodCallHandler(new FanheroNativeUtilsPlugin());
  }

  @Override
  public void onMethodCall(MethodCall call, Result result) {
    if (call.method.equals("getPlatformVersion")) {
      result.success("Android " + android.os.Build.VERSION.RELEASE);
    } else {
      result.notImplemented();
    }
  }
}

File: android/src/main/kotlin/com/fanhero/fanhero_native_utils/FanheroNativeUtilsPlugin.kt
package com.fanhero.fanhero_native_utils

import android.app.Activity
import android.content.pm.ActivityInfo

import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import io.flutter.plugin.common.PluginRegistry.Registrar

class FanheroNativeUtilsPlugin: MethodCallHandler {
  private val activity: Activity

  companion object {
    @JvmStatic
    fun registerWith(registrar: Registrar) {
      val channel = MethodChannel(registrar.messenger(), "fanhero_native_utils")
      channel.setMethodCallHandler(FanheroNativeUtilsPlugin(activity = registrar.activity()))
    }
  }

  constructor(activity: Activity) {
    this.activity = activity
  }

  override fun onMethodCall(call: MethodCall, result: Result) {
    if (call.method === "forcePortraitOrientation") {
      activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT;
    } else {
      result.notImplemented()
    }
  }
}
